From 83d70a1f4f437b3fb1c97192f19774427b9b53f5 Mon Sep 17 00:00:00 2001
From: Neta Zur <neta@marvell.com>
Date: Thu, 6 Aug 2015 10:30:05 +0300
Subject: [PATCH 0309/1240] dts: ap806: flc: add new dts apn-806-db-flc.dts

Update arch/arm/dts/apn-806-db.dts to dtsi file
create 2 new files that inherit  from the arch/arm/dts/apn-806-db.dtsi:
1. apn-806-db.dts - for regular db configuration
2. apn-806-db-flc.dts - for FLC mode add CCU dram window and RFU BootROM and SPI windows

Change-Id: Ie81f3576d4f890b686f5b049af03e7c553d9d80c
Signed-off-by: Neta Zur <neta@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/22586
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
---
 arch/arm/dts/Makefile           |   1 +
 arch/arm/dts/apn-806-db-flc.dts |  46 +++++++++++++++++
 arch/arm/dts/apn-806-db.dts     |  90 +--------------------------------
 arch/arm/dts/apn-806-db.dtsi    | 108 ++++++++++++++++++++++++++++++++++++++++
 4 files changed, 156 insertions(+), 89 deletions(-)
 create mode 100644 arch/arm/dts/apn-806-db-flc.dts
 create mode 100644 arch/arm/dts/apn-806-db.dtsi

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 19d8e9d..3d1abb4 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -55,6 +55,7 @@ dtb-$(CONFIG_MVEBU) += armada-385-db.dtb
 dtb-$(CONFIG_MVEBU) += armada-lp-db.dtb
 dtb-$(CONFIG_MVEBU) += armada-lp-palladium.dtb
 dtb-$(CONFIG_MVEBU) += apn-806-db.dtb
+dtb-$(CONFIG_MVEBU) += apn-806-db-flc.dtb
 
 targets += $(dtb-y)
 
diff --git a/arch/arm/dts/apn-806-db-flc.dts b/arch/arm/dts/apn-806-db-flc.dts
new file mode 100644
index 0000000..c3eba86
--- /dev/null
+++ b/arch/arm/dts/apn-806-db-flc.dts
@@ -0,0 +1,46 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2015 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+/dts-v1/;
+
+#include "apn-806-db.dtsi"
+
+/ {
+	model = "Marvell APN-806 Development Board FLC mode";
+	compatible = "marvell,apn-806-db-flc", "marvell,apn-806-db", "marvell,apn-806";
+
+	soc {
+		internal-regs {
+
+			ccu {
+				windows = <0x0 0x0  0x0 0xeff00000 0x3 /* DRAM window */
+				0x0 0xf2000000 0x0 0xe000000 0x00 /* IO window */>;
+			};
+
+			rfu {
+				windows = <
+					0x0 0xfff00000 0x0 0x0 0x0 /*BootROM */
+					0x0 0xf8000000 0x0 0x1100000 0x2 /* SPI */
+					0x0 0xfe000000 0x0 0x1f00000 0x05 /* PCIe window 31Mb for PCIe port*/
+					0x0 0xffe00000 0x0 0x100000 0x03 /* PCI-REG window 64Kb for PCIe-reg*/>;
+			};
+
+		};
+	};
+};
+
diff --git a/arch/arm/dts/apn-806-db.dts b/arch/arm/dts/apn-806-db.dts
index 6591eb1..7dc73fb 100644
--- a/arch/arm/dts/apn-806-db.dts
+++ b/arch/arm/dts/apn-806-db.dts
@@ -18,93 +18,5 @@
 
 /dts-v1/;
 
-#include "apn-806.dtsi"
-
-/ {
-	model = "Marvell APN-806 Development Board";
-	compatible = "marvell,apn-806-db", "marvell,apn-806";
-
-	#address-cells = <2>;
-	#size-cells = <2>;
-
-	memory {
-		device_type = "memory";
-		reg = <0x00000000 0x10000000 0 0>; /* 256 MB */
-	};
-	/* This should go only into devel boards */
-	sar {
-		compatible = "marvell,sample-at-reset";
-		#address-cells = <1>;
-		#size-cells = <0>;
-		reg = <0x4c 0x4d 0x0>;
-		chip_count = <2>;
-		bit_width = <5>;
-
-		freq {
-			key = "freq";
-			description = "CPU/DDR frequencies";
-			start-bit = <0>;
-			bit-length = <5>;
-			option-cnt = <4>;
-			options = "0x4", "2500/525 Mhz",
-				"0x6", "2500/800 Mhz",
-				"0x9", "2200/800 Mhz",
-				"0x13", "1600/800 Mhz";
-			default = <0x9>;
-			swap-bit;
-			status = "okay";
-		};
-		bootsrc {
-			key = "bootsrc";
-			description = "Boot source options";
-			start-bit = <5>;
-			bit-length = <2>;
-			option-cnt = <3>;
-			options = "0x0", "BootRoom",
-				"0x1", "Companion Device",
-				"0x2", "SPI";
-			default = <0x2>;
-			status = "okay";
-		};
-		c2c_ihb {
-			key = "c2c_ihb";
-			description = "C2C IHB select";
-			start-bit = <7>;
-			bit-length = <1>;
-			option-cnt = <2>;
-			options = "0x0", "IHB",
-				"0x1", "C2C";
-			default = <0x1>;
-			status = "okay";
-		};
-		dual_ap {
-			key = "dual_ap";
-			description = "Dual AP mode";
-			start-bit = <8>;
-			bit-length = <1>;
-			option-cnt = <2>;
-			options = "0x0", "Single AP mode",
-				"0x1", "Dual AP mode";
-			default = <0x0>;
-			status = "okay";
-		};
-		avs_enable {
-			key = "avs_enable";
-			description = "AVS Enable";
-			start-bit = <9>;
-			bit-length = <1>;
-			option-cnt = <2>;
-			options = "0x0", "AVS disabled",
-				"0x1", "AVS enabled";
-			default = <0x1>;
-			status = "okay";
-		};
-	};
-};
-
-&pinctl {
-		  /* 0 1 2 3 4 5 6 7 8 9 */
-	pin-func = < 1 1 1 1 1 1 1 1 1 0
-		     0 0 0 1 0 0>;
-};
+#include "apn-806-db.dtsi"
 
diff --git a/arch/arm/dts/apn-806-db.dtsi b/arch/arm/dts/apn-806-db.dtsi
new file mode 100644
index 0000000..7f3a82f
--- /dev/null
+++ b/arch/arm/dts/apn-806-db.dtsi
@@ -0,0 +1,108 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2015 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+#include "apn-806.dtsi"
+
+/ {
+	model = "Marvell APN-806 Development Board";
+	compatible = "marvell,apn-806-db", "marvell,apn-806";
+
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	memory {
+		device_type = "memory";
+		reg = <0x00000000 0x10000000 0 0>; /* 256 MB */
+	};
+	/* This should go only into devel boards */
+	sar {
+		compatible = "marvell,sample-at-reset";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x4c 0x4d 0x0>;
+		chip_count = <2>;
+		bit_width = <5>;
+
+		freq {
+			key = "freq";
+			description = "CPU/DDR frequencies";
+			start-bit = <0>;
+			bit-length = <5>;
+			option-cnt = <4>;
+			options = "0x4", "2500/525 Mhz",
+				"0x6", "2500/800 Mhz",
+				"0x9", "2200/800 Mhz",
+				"0x13", "1600/800 Mhz";
+			default = <0x9>;
+			swap-bit;
+			status = "okay";
+		};
+		bootsrc {
+			key = "bootsrc";
+			description = "Boot source options";
+			start-bit = <5>;
+			bit-length = <2>;
+			option-cnt = <3>;
+			options = "0x0", "BootRoom",
+				"0x1", "Companion Device",
+				"0x2", "SPI";
+			default = <0x2>;
+			status = "okay";
+		};
+		c2c_ihb {
+			key = "c2c_ihb";
+			description = "C2C IHB select";
+			start-bit = <7>;
+			bit-length = <1>;
+			option-cnt = <2>;
+			options = "0x0", "IHB",
+				"0x1", "C2C";
+			default = <0x1>;
+			status = "okay";
+		};
+		dual_ap {
+			key = "dual_ap";
+			description = "Dual AP mode";
+			start-bit = <8>;
+			bit-length = <1>;
+			option-cnt = <2>;
+			options = "0x0", "Single AP mode",
+				"0x1", "Dual AP mode";
+			default = <0x0>;
+			status = "okay";
+		};
+		avs_enable {
+			key = "avs_enable";
+			description = "AVS Enable";
+			start-bit = <9>;
+			bit-length = <1>;
+			option-cnt = <2>;
+			options = "0x0", "AVS disabled",
+				"0x1", "AVS enabled";
+			default = <0x1>;
+			status = "okay";
+		};
+	};
+};
+
+&pinctl {
+		  /* 0 1 2 3 4 5 6 7 8 9 */
+	pin-func = < 1 1 1 1 1 1 1 1 1 0
+		     0 0 0 1 0 0>;
+};
+
-- 
1.9.1

